homeassistant/components/somfy_mylink/cover.py

```diff
@@ -2,49 +2,66 @@
 import logging
 
 from homeassistant.components.cover import (
+    DEVICE_CLASS_BLIND,
+    DEVICE_CLASS_SHUTTER,
     DEVICE_CLASS_WINDOW,
     ENTITY_ID_FORMAT,
     CoverEntity,
 )
+from homeassistant.const import STATE_CLOSED, STATE_OPEN
+from homeassistant.helpers.restore_state import RestoreEntity
 from homeassistant.util import slugify
 
-from . import CONF_DEFAULT_REVERSE, DATA_SOMFY_MYLINK
+from .const import (
+    CONF_DEFAULT_REVERSE,
+    CONF_REVERSE,
+    DATA_SOMFY_MYLINK,
+    DOMAIN,
+    MANUFACTURER,
+    MYLINK_ENTITY_IDS,
+    MYLINK_STATUS,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
+MYLINK_COVER_TYPE_TO_DEVICE_CLASS = {0: DEVICE_CLASS_BLIND, 1: DEVICE_CLASS_SHUTTER}
+
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_entry(hass, config_entry, async_add_entities):
     """Discover and configure Somfy covers."""
-    if discovery_info is None:
-        return
-    somfy_mylink = hass.data[DATA_SOMFY_MYLINK]
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    mylink_status = data[MYLINK_STATUS]
+    somfy_mylink = data[DATA_SOMFY_MYLINK]
+    mylink_entity_ids = data[MYLINK_ENTITY_IDS]
     cover_list = []
-    try:
-        mylink_status = await somfy_mylink.status_info()
-    except TimeoutError:
-        _LOGGER.error(
-            "Unable to connect to the Somfy MyLink device, "
-            "please check your settings"
-        )
-        return
+
     for cover in mylink_status["result"]:
         entity_id = ENTITY_ID_FORMAT.format(slugify(cover["name"]))
-        entity_config = discovery_info.get(entity_id, {})
-        default_reverse = discovery_info[CONF_DEFAULT_REVERSE]
+        mylink_entity_ids.append(entity_id)
+
+        entity_config = config_entry.options.get(entity_id, {})
+        default_reverse = config_entry.options.get(CONF_DEFAULT_REVERSE)
+
         cover_config = {}
         cover_config["target_id"] = cover["targetID"]
```

Comment:
The per-entity config is stored in the config_entry options based on the "entity_id" (which is the name reported from the API), which presumably could change and break the option storage. Given that unique_id is based upon targetID would it be more stable to use targetID for that config_entry storage?

Reply:
I think that makes sense to do after YAML support is dropped in the future.  Since we want to import the existing configuration, I think it makes sense to keep it as-is for now.